FaunaDB.Client.Test/ParallelTest.cs

```diff
@@ -0,0 +1,113 @@
+﻿using NUnit.Framework;
+using System;
+using System.Collections.Generic;
+using FaunaDB.Types;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using static FaunaDB.Query.Language;
+using System.Collections.Concurrent;
+using FaunaDB.Client;
+
+namespace Test
+{
+    public class ParallelTest: TestCase
+    {
+        private const int IN_PARALLEL = 1000;
+        private const int MAX_ATTEMPTS = 10;
+        private const string COLLECTION_NAME = "ParallelTestCollection";
+        
+        [OneTimeSetUp]
+        public void SetUpCollection()
+        {
+            SetUpCollectionAsync().Wait();
+        }
+        public async Task SetUpCollectionAsync()
+        {
+            var client = new FaunaClient(secret: faunaSecret, endpoint: faunaEndpoint);
+            Value exist = await client.Query(Exists(Collection(COLLECTION_NAME)));
```

Comment:
Instead of casting to bool using `(bool)` as you did below, let's do it this way:
```suggestion
            bool exist = (await client.Query(Exists(Collection(COLLECTION_NAME)))).To<bool>().Value;
```